src/main/java/core/basesyntax/service/impl/TransactionProcessorServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionProcessorService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class TransactionProcessorServiceImpl implements TransactionProcessorService {
+    private Map<String, Integer> fruitCounts;
+    private final Map<FruitTransaction.Operation, OperationStrategy> strategies;
+
+    public TransactionProcessorServiceImpl(Map<FruitTransaction.Operation,
+            OperationStrategy> strategies) {
+        this.strategies = strategies;
+    }
+
+    @Override
+    public Map<String, Integer> processTransaction(List<FruitTransaction> transactions) {
+        fruitCounts = Storage.getFruits();
+        if (transactions != null) {
+            for (FruitTransaction transaction : transactions) {
+                processSingleTransaction(transaction);
+            }
+        }
+        return fruitCounts;
+    }
+
+    public void processSingleTransaction(FruitTransaction transaction) {
+        FruitTransaction.Operation typeOfOperation = transaction.getType();
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+
+        OperationStrategy strategy = strategies.get(typeOfOperation);
+        if (strategy != null) {
+            strategy.apply(fruitCounts, fruit, quantity);
+
+        } else {
+            System.out.println("Invalid operation type: " + typeOfOperation);
```

Comment:
this is not a way to go, printlns should be removed from "production" code, we can't rely on them. If this is not an expected behaviour better to be upfront about it and 'fail fast' than 'eat it' silently